### ISSUE TITLE
Return exit code

### ISSUE THREAD

**Issue by royjs** [databus23/helm-diff#54]
I am deploying with Helm from CI and it would be really helpful to have the commands return a different exit code based on whether there is some changes or not. From CI I usually don't care about the actual changes but I want to know if something changed or not.

My suggestion would be to have a `--detailed-exit-code` flag similar to `terraform plan` or `git diff`

The best would probably to do like terraform and return something like this:

- 0 if no error and no diff
- 1 if there is an error
- 2 if no error but a diff

This change would also allow to implement the same flag in `helmfile`. See roboll/helmfile#133

Any thoughts?

**Comment by travisgroth** [comment]
I'd like to see this as well so that a helm deployment is only triggered if there is an actual difference.   The lack of information as to whether a deployment is needed has been our biggest issue with helm and integrating it into more 'driverless' automation.

The exit code is actually easy to implement in helm-diff based on my testing, but helm itself will treat any non-zero exit code as an error.  So, this should probably be coupled with a change in helm to define an exit code contract to signal failure versus a code that should be passed back to the caller.

**Comment by royjs** [comment]
@travisgroth thanks for looking into this, I didn't have time to do it yet.
It looks like it is a little more complicated than what I thought. 
Were you planning on opening an issue in the helm project ? I can also do it if you prefer

**Comment by travisgroth** [comment]
@royjs it probably makes sense for you to initiate the issues across the 3 projects, but I'll chime in on the discussion.   Assuming there is an agreement on the path forward and the interface, I may be able to do a PR for one or more of these since I've already wandered through the code path.

**Comment by mumoshu** [comment]
@travisgroth @royjs @cmeury Would you be able to participate in the helm part of this issue at https://github.com/helm/helm/issues/4170?  As a workaround, I was trying to bypass helm's plugin system, hence call `helm-diff` directly. But turned out it isn't very feasible as explained in the issue. I'm now eager to implement this in helm. Appreciate your support there 😃

**Comment by mumoshu** [comment]
https://github.com/helm/helm/pull/4367 was merged to the upstream to resolve https://github.com/helm/helm/issues/4170.

It should be good to start returning the separate exit code for no error but diff now.

**Comment by mumoshu** [comment]
#78 in combination with `helm` built from the current master seems to work:

```
$ helm211dev diff upgrade myrelease stable/mysql --detailed-exitcode
...
$ echo $?
2
```

@royjs @travisgroth @cmeury Could you verify that it works for you?

For your reference, `helm` can be build from master by running `make build && cp bin/helm path/to/your/bin/helm211dev`.

**Comment by mumoshu** [comment]
I'm not going to close this until helm 2.11 is released. It may take a few months later, because helm is still 2.10.rc-1 today.

**Comment by mumoshu** [comment]
@royjs Hey. Thanks for the report! Yes, and it should be fixed by #79.

**Comment by mumoshu** [comment]
FYI, helm 2.10 has already been released with the feature.

**Comment by mumoshu** [comment]
I'm going to closing this after #81, bumping helm dependency to 2.10

**Comment by mumoshu** [comment]
Released v2.9.0+3 and v2.10.0+1 that includes this feature. Closing as resolved, but please feel free to try and report any issue! Thanks.
